internal/pkg/cli/task_run.go

```diff
@@ -248,22 +254,41 @@ func (o *runTaskOpts) askTaskGroupName() error {
 	return nil
 }
 
+func (o *runTaskOpts) askAppName() error {
+	if o.AppName() != "" {
+		return nil
+	}
+
+	// NOTE: if we are not in a workspace and app flag is not specified, prompt the user to select an application or "None"
+	app, err := o.sel.Application(fmtTaskRunAppPrompt, taskRunAppPromptHelp, appEnvOptionNone)
```

Comment:
One thing that I'm not clear about is which flags to provide for the `task.DefaultVPCRunner` codepath. If the user doesn't provide `--app`, `--env`, or `--subnets` then we will prompt them. Is there a way of avoiding the prompt for the `DefaultVPCRunner` route?

Reply:
I guess there is this tradeoff between making sure that the user really really wants to run in "default" and making assumptions to some degree. 

Originally, if they are not in a workspace and they don't provide a value for `--app`, then we assume that they want to run in default. Since we don't want to make this assumption, we now prompt for "application" and include the `None` option. 

Since we want an explicit answer from the user that they want to run in "default", I think a (cheap) way is to have one more flag called "--default".

Reply:
Sounds good, so the minimal flag setting that users need to provide is `copilot task run -n myGroup --default` :+1: 

Also do you think we should change:
```go
// From
cmd.Flags().StringVar(&vars.dockerfilePath, dockerFileFlag, "", dockerFileFlagDescription)

// To
cmd.Flags().StringVar(&vars.dockerfilePath, dockerFileFlag, defaultDockerfilePath, dockerFileFlagDescription)
```

Reply:
I will make a separate PR for `--default` flag because that requires some changes to `Validate`, `flag.go`, etc.